app/javascript/channels/appearance_channel.js

```diff
@@ -0,0 +1,47 @@
+import consumer from "./consumer";
+
+consumer.subscriptions.create("AppearanceChannel", {
+  connected() {
+    // Called when the subscription is ready for use on the server
+  },
+
+  disconnected() {
+    // Called when the subscription has been terminated by the server
+  },
+
+  received(data) {
+    const { users } = data;
+    users.map((user) => {
+      const user_id = user.id;
+      const isUserOnline = user.online;
+      const currentItem = $(`#user-item[data-user-id="${user_id}"]`);
+      const item = `<a class="item" id="user-item" data-user-id="${user_id}">
+                        ${user.username}
+                    </a>`;
+      const onlineFlag = `<a class="ui mini green empty circular label" id="online-flag" />`;
+      const currentOnlineFlag = $(
+        `#user-item[data-user-id="${user_id}"] > a#online-flag`
+      );
+
+      if (!isUserOnline && currentItem.length < 1) {
+        $("#user-container").append(item);
+      }
+
+      if (isUserOnline && currentItem.length < 1) {
+        $("#user-container").append($(item).append(onlineFlag));
+      }
+
+      if (
+        isUserOnline &&
+        currentItem.length > 0 &&
+        currentOnlineFlag.length < 1
+      ) {
+        $(currentItem).append(onlineFlag);
+      }
+
+      if (!isUserOnline) {
+        $(currentOnlineFlag).remove();
+      }
```

Comment:
đoạn này t nghĩ viết lại như thế này thì đọc logic nó liền mạch hơn, như ông viết bị rời rạc quá.
```suggestion
      if (!currentItem.length) {
        if (isUserOnline) {
          $(item).append(onlineFlag);
        }
        $("#user-container").append(item);
      } else {
        if (!isUserOnline) {
          $(currentOnlineFlag).remove();
        } else if (!currentOnlineFlag.length) {
          $(currentItem).append(onlineFlag);
        }
      }
```